### ISSUE TITLE
traceback from script module with old style var (when used with include?)

### ISSUE THREAD

**Issue by jphalip** [ansible/ansible#4256]
Consider the following simple example:

main.yml:

```
- hosts: all
  vars:
    myvar: /home
  tasks:
    - include: included.yml
```

included.yml:

```
- script: myscript.sh $myvar
```

myscript.sh:

```
ls $1
```

Running main.yml raises an exception:

```
TASK: [script myscript.sh /home] ********************************************** 
fatal: [111.111.111.111] => Traceback (most recent call last):
  File "/Users/julien/.virtualenvs/myenv/lib/python2.6/site-packages/ansible/runner/__init__.py", line 382, in _executor
    exec_rc = self._executor_internal(host, new_stdin)
  File "/Users/julien/.virtualenvs/myenv/lib/python2.6/site-packages/ansible/runner/__init__.py", line 471, in _executor_internal
    return self._executor_internal_inner(host, self.module_name, self.module_args, inject, port, complex_args=complex_args)
  File "/Users/julien/.virtualenvs/myenv/lib/python2.6/site-packages/ansible/runner/__init__.py", line 654, in _executor_internal_inner
    result = handler.run(conn, tmp, module_name, module_args, inject, complex_args)
  File "/Users/julien/.virtualenvs/myenv/lib/python2.6/site-packages/ansible/runner/action_plugins/script.py", line 45, in run
    source = utils.path_dwim_relative(inject['_original_file'], 'files', source, self.runner.basedir)
  File "/Users/julien/.virtualenvs/myenv/lib/python2.6/site-packages/ansible/utils/__init__.py", line 259, in path_dwim_relative
    if os.path.exists(source2):
  File "/Users/julien/.virtualenvs/myenv/lib/python2.6/genericpath.py", line 18, in exists
    st = os.stat(path)
TypeError: stat() argument 1 must be encoded string without NULL bytes, not str
```

Also, this error is raised only if the script requires an argument. If no argument is used then the script runs fine without error.

Any idea why this is happening and how to fix it?

**Comment by jimi-c** [comment]
The variable should be specified as `{{ myvar }}`, not `$myvar`.

**Comment by mpdehaan** [comment]
I don't think that's valid though... yes, we want people using new style variables, but even though that is the deprecated variable syntax it should definitely _work_

**Comment by jphalip** [comment]
I've tried with the new-style variable and the same exception is still raised.

**Comment by jphalip** [comment]
It is the script's name (`myscript.sh`) that causes the exception to be raised by `os.stat(path)` as it is of `str` type.

**Comment by jphalip** [comment]
So I've done some digging. This has in fact nothing to do with old-style vs new-style variables.

The problem is in Python itself. It's a bug with the `shlex` and `StringIO` modules, which appears to have been fixed in 2.7 but not 2.6:
http://bugs.python.org/issue6988
http://hg.python.org/cpython/rev/27ae7d4e1983/

The problem is that `shlex.split()` returns tokens in an encoded format, which makes `os.path.exists()` crash (via the call to `path_dwim_relative()`).

I've posted a work-around in https://github.com/ansible/ansible/pull/4269 to ensure it works in Python 2.6 as well as 2.7.
